pkg/run/storage.go

```diff
@@ -64,6 +68,40 @@ func (r *RunStorageService) ensureBucketExists(ctx context.Context, bucket strin
 	return err
 }
 
+func (r *RunStorageService) triggerBucketNotification(ctx context.Context, bucket string, key string, eventType v1.BucketNotificationType) error {
+	evt := &v1.TriggerRequest{
+		Context: &v1.TriggerRequest_Notification{
+			Notification: &v1.NotificationTriggerContext{
+				Source: bucket,
+				Notification: &v1.NotificationTriggerContext_Bucket{
+					Bucket: &v1.BucketNotification{
+						Key:  key,
+						Type: eventType,
+					},
+				},
+			},
+		},
+	}
+
+	worker, err := r.pool.GetWorker(&pool.GetWorkerOptions{
+		Trigger: evt,
+		Filter: func(w worker.Worker) bool {
+			_, ok := w.(*worker.BucketNotificationWorker)
+			return ok
+		},
+	})
+	if err != nil {
+		// There is no worker for this notification
+		return nil
+	}
+
+	if _, err := worker.HandleTrigger(ctx, evt); err != nil {
+		return fmt.Errorf("error occcured triggering bucket notification: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Should probably use GetWorkers here similar to event publishing to make sure all workers recieve all events.

Reply:
These should also be sent as non-blocking requests, currently triggering an API that writes a file locally will wait for all workers to be triggered. This behaviour should be asynchronous.

Reply:
> Should probably use GetWorkers here similar to event publishing to make sure all workers recieve all events.

There should really only be one worker that is triggered since overlapping events are illegal